service/chain_service.go

```diff
@@ -188,6 +189,11 @@ func (cs *chainService) previousHeader(bs *domains.BlockHeaderSource) (*domains.
 func (cs *chainService) insert(h *domains.BlockHeader) (*domains.BlockHeader, error) {
 	err := cs.Repositories.Headers.AddHeaderToDatabase(*h)
 	if err != nil {
+		if errors.Is(err, customErrs.NewUniqueViolationError()) {
```

Comment:
I don't understand how this code would prevent multiple webhook calls after `insert()`

Reply:
fixed

Reply:
I believe we don't need this check here. Please move warning to Add() method

Reply:
Moved to Add method